### ISSUE TITLE
Custom blocks don't cut-paste properly

### ISSUE THREAD

**Issue by superkelvint** [TypeCellOS/BlockNote#813]
**Describe the bug**
When cutting-pasting custom blocks, unexpected behavior happens. 

In the case of the custom alert block from https://www.blocknotejs.org/docs/custom-schemas/custom-blocks, the alert box seems to disappear. In other cases, indeterminate behavior happens.

Copying and pasting seems to work. Its cutting and pasting that doesn't.

**To Reproduce**
Please watch the screencast here: https://share.vidyard.com/watch/x2VNexUuFWMZy2X7kLc7ym

Basically, select the custom block, then "ctrl-x", "ctrl-v".

**Comment by superkelvint** [comment]
Actually, with another custom block i was building, copying/pasting didn't work either.

**Comment by matthewlipski** [comment]
Managed to reproduce on Chrome:

https://github.com/TypeCellOS/BlockNote/assets/50169049/20b3b2ca-056a-4b95-a711-14ed5bcdfe6f

Seems super flaky though, could only do it on a fresh reload of the page

**Comment by superkelvint** [comment]
I'm using Chrome 122 on Ubuntu Linux.

**Comment by thedevangvishnu** [comment]
The issue is even more weird. If you copy the blocks and paste them, it works as expected. And then if you cut and paste those blocks, IT WORKS. 

But if you cut-paste the first time, it doesn't. But when a block/group of blocks have been already copy-pasted, then cut-paste seem to work.


https://github.com/TypeCellOS/BlockNote/assets/113965202/9cfe579a-d05b-4abf-8453-e552ea0eea2a
